torch_xla/csrc/tensor.cpp

```diff
@@ -284,6 +284,26 @@ void XLATensor::SetXlaData(torch::lazy::BackendDataPtr handle, bool sync) {
   }
 }
 
+void XLATensor::CreateShardedIrValue(const ShardingSpecPtr sharding_spec) {
+  torch::lazy::Value old_value = GetIrValue();
+  XLA_CHECK(old_value && old_value.node != nullptr)
+      << "Cannot create a sharded IR value if an IR value does not already "
+         "exist";
+  torch::lazy::Value new_ir = dynamic_cast<XlaNode*>(old_value.node.get())
+                                  ->CloneWithSharding(sharding_spec->sharding);
+  data()->ir_value = std::move(new_ir);
```

Comment:
some node's operands could be pointing to the old_ir.

Reply:
Yeah I think this is the root cause of the problem I was seeing before. If I can keep the operands up to date, the clone should work smoothly.